src/Compilers/CSharp/Portable/Compiler/MethodBodySynthesizer.Lowered.cs

```diff
@@ -277,4 +285,329 @@ internal static BoundBlock ConstructSingleInvocationMethodBody(SyntheticBoundNod
                         : F.Block(F.Return(invocation));
         }
     }
+
+    internal class SynthesizedAsIntValueMethod : SynthesizedGlobalMethodSymbol
+    {
+        private readonly Emit.PEModuleBuilder _module;
+
+        public SynthesizedAsIntValueMethod(Emit.PEModuleBuilder module, PrivateImplementationDetails pi)
+            : base(module.SourceModule, pi,
+                   module.Compilation.GetSpecialType(SpecialType.System_Boolean),
```

Comment:
This doesn't report use-site errors, are we certain we checked this type before? Please add a comment to that effect.